packages/lesswrong/components/ultraFeed/UltraFeedCommentsDialog.tsx

```diff
@@ -0,0 +1,189 @@
+  dialogContent: {
+    padding: 20,
+    paddingTop: 0,
+    [theme.breakpoints.down('sm')]: {
+      padding: 10,
+      paddingTop: 0,
+    }
+  },
+  titleContainer: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    paddingLeft: 16,
+    paddingRight: 16,
+    paddingTop: 8,
+    marginBottom: 8,
+  },
+  title: {
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    fontSize: "1.3rem",
+    fontWeight: 600,
+  },
+  closeButton: {
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    color: theme.palette.grey[500],
+    fontWeight: 600,
+    fontSize: "1.1rem",
+    cursor: 'pointer',
+    padding: 8,
+    '&:hover': {
+      color: theme.palette.grey[700],
+    },
+    
+  },
+  dialogPaper: {
+    maxWidth: 765,
+    margin: 4
+  },
+  scrolledHighlight: {
+    backgroundColor: `${theme.palette.secondary.light}6c`,
+  }
+}));
+
+const UltraFeedCommentsDialog = ({
+  document,
+  collectionName,
+  onClose
+}: {
+  document: PostsListWithVotes | UltraFeedComment,
+  collectionName: "Posts" | "Comments",
+  onClose: () => void,
+}) => {
+  const { LWDialog, CommentsListSection, Loading } = Components;
+  const classes = useStyles(styles);
+
+  const isPost = collectionName === "Posts";
+  const comment = isPost ? null : (document as UltraFeedComment);
+  const postId = isPost ? document._id : comment?.postId;
+  const topLevelCommentId = comment?.topLevelCommentId ?? comment?._id;
+  const postTitle = isPost ? document.title : comment?.post?.title;
+  const targetCommentId = !isPost ? document._id : undefined;
+
+  const { document: postDataForTree, loading: loadingPost } = useSingle({
+    documentId: postId,
+    collectionName: "Posts",
+    fragmentName: "PostsDetails",
+    skip: !postId,
+  });
+
+  const { results: postComments, loading: loadingPostComments, totalCount: postCommentsTotalCount } = useMulti({
+    terms: {
+      view: "postCommentsTop",
+      postId,
+      limit: 50, // Consider pagination later if needed
+    },
+    collectionName: "Comments",
+    fragmentName: "CommentsList",
+    skip: !isPost || !postId,
+    enableTotal: true,
+  });
+
+  const { results: threadComments, loading: loadingThreadComments, totalCount: threadCommentsTotalCount } = useMulti({
+    terms: {
+      view: "repliesToCommentThreadIncludingRoot",
+      topLevelCommentId,
+      limit: 50, // Fetch a large number to get the whole thread initially
+    },
+    collectionName: "Comments",
+    fragmentName: "CommentsList",
+    skip: isPost || !topLevelCommentId, // Only run if collectionName is Comments and we have the ID
+    enableTotal: true,
+  });
+
+  const isLoading = loadingPost || (isPost ? loadingPostComments : loadingThreadComments);
+  const comments = isPost ? postComments : threadComments;
+  const totalCount = isPost ? postCommentsTotalCount : threadCommentsTotalCount;
+
+  // scroll to comment clicked on when dialog opens
+  useEffect(() => {
```

Comment:
this seems a bit complicated.  if we're rendering `CommentsNode`s somewhere down the tree (which I bet we are, if we're just using `CommentsListSection` directly), they already have functionality to handle scrolling to a specific comment via `useCommentLinkState`.